src/id-data-controls.tsx

```diff
@@ -0,0 +1,271 @@
+                                fields.splice(idx, 1);
+                            })
+                        );
+                    }}
+                    onChange={(idx, new_data) =>
+                        setCustomIdData(
+                            produce((fields) => {
+                                if (!fields[idx]) throw new Error('index out of bounds');
+                                fields[idx] = new_data;
+                            })
+                        )
+                    }
+                    fields={custom_id_data}
+                    title={t('saved-data', 'id-data-controls')}
+                    hasPrimary={false}>
+                    <IntlProvider scope="id-data-controls" definition={window.I18N_DEFINITION}>
+                        <div>
+                            <p>
+                                <MarkupText id="saved-data-explanation" />
+                            </p>
+                            <div className="form-group">
+                                <input
+                                    type="checkbox"
+                                    id="always-fill-in"
+                                    className="form-element"
+                                    checked={SavedIdData.shouldAlwaysFill()}
+                                    onChange={(event) => {
+                                        SavedIdData.setAlwaysFill(!SavedIdData.shouldAlwaysFill());
+                                    }}
+                                />
+                                <label htmlFor="always-fill-in">
+                                    <Text id="always-fill-in" />
+                                </label>
+                            </div>
+
+                            <div className="form-group" style="width: 100%; display: table;">
+                                <div className="col40">
+                                    <strong>
+                                        <label htmlFor="name-input-value-fixed-id-data">{t('name', 'generator')}</label>
+                                    </strong>
+                                </div>
+                                <div className="col60" style="padding-left: 10px; box-sizing: border-box;">
+                                    <InputControl
+                                        id="name-input"
+                                        suffix="fixed-id-data"
+                                        onChange={(v) => handleFixedChange('name', v)}
+                                        value={fixed_id_data.name}
+                                    />
+                                </div>
+                                <div className="clearfix" />
+
+                                <div className="col40">
+                                    <strong>
+                                        <label htmlFor="email-input-value-fixed-id-data">
+                                            {t('email-address', 'generator')}
```

Comment:
Can't this be a `<Text>`?

(also below)

Reply:
No, because they are in a different scope